runtime/vm/swalk.c

```diff
@@ -121,447 +121,164 @@ maxFramesOnStack(J9StackWalkState *walkState)
-#else /* J9VM_INTERP_STACKWALK_TRACING */
-	Trc_VM_WalkStackFrames_Entry(currentThread, walkState->walkThread, walkState->flags,
-			walkState->walkThread->sp, walkState->walkThread->arg0EA,
-			walkState->walkThread->pc, walkState->walkThread->literals,
-			walkState->walkThread->entryLocalStorage, walkState->walkThread->j2iFrame);
-#endif /* J9VM_INTERP_STACKWALK_TRACING */
-
-	if (J9_ARE_ANY_BITS_SET(walkState->flags, J9_STACKWALK_RESUME)) {
-		if (NULL != walkState->jitInfo) {
-			goto resumeJitWalk;
-		}
-		walkState->flags &= ~J9_STACKWALK_RESUME;
-		goto resumeInterpreterWalk;
-	}
-
-	walkState->loopBreaker = 0;
-	if (NULL != currentThread) {
-		oldState = currentThread->activeWalkState;
-		currentThread->activeWalkState = walkState;
-		if (J9_ARE_ANY_BITS_SET(currentThread->privateFlags2, J9_PRIVATE_FLAGS2_ASYNC_GET_CALL_TRACE)) {
-			/* Add one because walkFrames decrements loopBreaker and bails when it reaches 0 */
-			walkState->loopBreaker = maxFramesOnStack(walkState) + 1;
-		}
-	}
-
-	walkState->javaVM = walkState->walkThread->javaVM;
+	walkState->javaVM = targetThread->javaVM;
 	walkState->currentThread = currentThread;
+	walkState->walkThread = targetThread;
 	walkState->cache = NULL;
 	walkState->framesWalked = 0;
 	walkState->previousFrameFlags = 0;
-	walkState->arg0EA = walkState->walkThread->arg0EA;
-	walkState->pcAddress = &(walkState->walkThread->pc);
-	walkState->pc = walkState->walkThread->pc;
+	walkState->arg0EA = targetThread->arg0EA;
+	walkState->pcAddress = &(targetThread->pc);
+	walkState->pc = targetThread->pc;
 	walkState->nextPC = NULL;
-	walkState->walkSP = walkState->walkThread->sp;
-	walkState->literals = walkState->walkThread->literals;
+	walkState->walkSP = targetThread->sp;
+	walkState->literals = targetThread->literals;
 	walkState->argCount = 0;
 	walkState->linearSlotWalker = NULL;
 	walkState->stackMap = NULL;
 	walkState->inlineMap = NULL;
 	walkState->inlinedCallSite = NULL;
+	walkState->privateFlags = targetThread->privateFlags;
+	walkState->stackObject = targetThread->stackObject;
+	walkState->stackWalkState = targetThread->stackWalkState;
 #ifdef J9VM_INTERP_NATIVE_SUPPORT
 	walkState->jitInfo = NULL;
 	walkState->inlineDepth = 0;
 	walkState->inlinerMap = NULL;
-	walkState->walkedEntryLocalStorage = walkState->walkThread->entryLocalStorage;
-	walkState->i2jState = walkState->walkedEntryLocalStorage ? &(walkState->walkedEntryLocalStorage->i2jState) : NULL;
-	walkState->j2iFrame = walkState->walkThread->j2iFrame;
+	J9VMEntryLocalStorage* els = targetThread->entryLocalStorage;
+	walkState->walkedEntryLocalStorage = els;
+	walkState->i2jState = NULL != els ? &(els->i2jState) : NULL;
+	walkState->j2iFrame = targetThread->j2iFrame;
+	walkState->jitGlobalStorageBase = NULL != els ? els->jitGlobalStorageBase : NULL;
+	walkState->jitFPRegisterStorageBase = NULL !=els ? els->jitFPRegisterStorageBase : NULL;
+	walkState->oldEntryLocalStorage = NULL != els ? els->oldEntryLocalStorage : NULL;
```

Comment:
brackets around the conditions
`walkState->i2jState = (NULL != els) ? &(els->i2jState) : NULL;`